tests/launchsecurity/sev.go

```diff
@@ -38,4 +191,47 @@ var _ = Describe("[sig-compute]AMD Secure Encrypted Virtualization (SEV)", func(
 		}, 30)
 		Expect(err).ToNot(HaveOccurred())
 	})
+
+	It("should run guest attestation", func() {
+		vmi := newSEVFedora(libvmi.WithSEVAttestation())
+		vmi = tests.RunVMI(vmi, 30)
+		Eventually(ThisVMI(vmi), 60).Should(BeInPhase(v1.Scheduled))
+
+		virtClient, err := kubecli.GetKubevirtClient()
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Fetching platform certificates")
+		sevPlatformInfo, err := virtClient.VirtualMachineInstance(vmi.Namespace).SEVFetchCertChain(vmi.Name)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(sevPlatformInfo).To(Equal(expectedSEVPlatformInfo))
+
+		By("Setting up session parameters")
+		err = virtClient.VirtualMachineInstance(vmi.Namespace).SEVSetupSession(vmi.Name, sevSessionOptions)
+		Expect(err).ToNot(HaveOccurred())
+		Eventually(ThisVMI(vmi), 60).Should(BeRunning())
+		tests.WaitForVMICondition(virtClient, vmi, v1.VirtualMachineInstancePaused, 30)
+
+		By("Querying launch measurement")
+		sevMeasurementInfo, err := virtClient.VirtualMachineInstance(vmi.Namespace).SEVQueryLaunchMeasurement(vmi.Name)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(sevMeasurementInfo.APIMajor).To(Equal(expectedSEVMeasurementInfo.APIMajor))
+		Expect(sevMeasurementInfo.APIMinor).To(Equal(expectedSEVMeasurementInfo.APIMinor))
+		Expect(sevMeasurementInfo.BuildID).To(Equal(expectedSEVMeasurementInfo.BuildID))
+		Expect(sevMeasurementInfo.Policy).To(Equal(expectedSEVMeasurementInfo.Policy))
+		Expect(sevMeasurementInfo.LoaderSHA).To(Equal(expectedSEVMeasurementInfo.LoaderSHA))
+		measure := verifyMeasurement(&sevMeasurementInfo, tikBase64)
+		sevSecretOptions := encryptSecret(diskSecret, measure, tikBase64, tekBase64)
+
+		By("Injecting launch secret")
+		err = virtClient.VirtualMachineInstance(vmi.Namespace).SEVInjectLaunchSecret(vmi.Name, sevSecretOptions)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Unpausing the VirtualMachineInstance")
+		err = virtClient.VirtualMachineInstance(vmi.Namespace).Unpause(vmi.Name, &v1.UnpauseOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		tests.WaitForVMIConditionRemovedOrFalse(virtClient, vmi, v1.VirtualMachineInstancePaused, 30)
```

Comment:
One thing that I don't quite understand here:
As far as I can tell the VMI doesn't contain any encrypted disks that require an injected secret in order to boot. Is this correct? If it is, how this test validates that the injection process is successful?

Please correct me if I'm missing something here

Reply:
cool :) thanks